target_chains/cosmwasm/tools/src/pyth-wrapper.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  ChainExecutor,
+  ExecuteContractResponse,
+} from "./chains-manager/chain-executor";
+import { ChainQuerier } from "./chains-manager/chain-querier";
+
+export type UpdateFeeResponse = {
+  denom: string;
+  amount: string;
+};
+
+export type Price = {
+  price: number;
+  conf: number;
+  expo: number;
+  publish_time: number;
+};
+
+export type PriceFeedResponse = {
+  id: string;
+  price: Price;
+  ema_price: Price;
+};
+
+export class PythWrapperQuerier {
+  constructor(private chainQuerier: ChainQuerier) {}
+
+  // get update fee
+  async getUpdateFee(
+    contractAddr: string,
+    vaas: string[]
+  ): Promise<UpdateFeeResponse> {
+    try {
+      const updateFeeResponse = await this.chainQuerier.getSmartContractState({
+        contractAddr,
+        query: {
+          get_update_fee: {
+            vaas,
+          },
+        },
+      });
+
+      return updateFeeResponse as UpdateFeeResponse;
+    } catch (e) {
+      throw new Error("Error fetching update fee");
+    }
+  }
+
+  // get price feed
+  async getPriceFeed(
+    contractAddr: string,
+    id: string
+  ): Promise<PriceFeedResponse> {
+    try {
+      const priceFeedResponse = await this.chainQuerier.getSmartContractState({
+        contractAddr,
+        query: {
+          price_feed: {
+            id,
+          },
+        },
+      });
+
+      return priceFeedResponse as PriceFeedResponse;
+    } catch (e) {
+      throw new Error("Error fetching update fee");
+    }
+  }
+}
+
+export type Fund = {
+  denom: string;
+  amount: string;
+};
```

Comment:
I think the combination of an amount and denom is usually called a `Coin` in cosmos?

Reply:
also (1) this type seems unused and (2) it's a duplicate of UpdateFeeResponse

Reply:
removing it